crates/edr_napi/test/solidity-tests.ts

```diff
@@ -76,6 +76,31 @@ describe("Solidity Tests", () => {
       new RegExp("Hex decoding error")
     );
   });
+
+  it("filters tests according to pattern", async function () {
+    const artifacts = [
+      loadContract("./artifacts/SetupConsistencyCheck.json"),
+    ];
+    // All artifacts are test suites.
+    const testSuites = artifacts.map((artifact) => artifact.id);
+
+    const results = await runAllSolidityTests(artifacts, testSuites, {
+      projectRoot: __dirname,
+      testPattern: 'Multiply',
+    });
+
+    assert.equal(results.length, artifacts.length);
```

Comment:
From an integration test perspective, this result would be the same as no filter being applied. I'd recommend adding at least one test that doesn't match all artifacts.

Reply:
Note that every artifact will produce a suite result, even if it contains zero test results, so this assertion will always be true for every filter config.